saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticationRequest.java

```diff
@@ -112,8 +146,18 @@ public static Builder builder() {
 		private List<Saml2X509Credential> credentials = new LinkedList<>();
 		private String destination;
 		private String assertionConsumerServiceUrl;
+		private boolean deflate = true;
+		private MultiValueMap<String, String> queryParameters = new LinkedMultiValueMap<>();
 
-		private Builder() {
+		private Builder(Saml2AuthenticationRequest request) {
```

Comment:
Instead of copying here, if you copied in the static method, then you wouldn't need the if guard. Also, it would align a bit more closely with `OAuth2AuthorizationRequest`.

Reply:
Same commit for this as the [previous comment](https://github.com/spring-projects/spring-security/pull/7758#discussion_r371527842)

I agree, please consider f5c9769b0cf5c941c6efd2509ad3b5d4bb46f087 as the solution